contracts/factory/MagicDropCloneFactory.sol

```diff
@@ -0,0 +1,245 @@
+    ==============================================================*/
+
+    struct MagicDropFactoryStorage {
+        mapping(bytes32 => bool) usedSalts;
+    }
+
+    /*==============================================================
+    =                            STORAGE                           =
+    ==============================================================*/
+
+    // keccak256(abi.encode(uint256(keccak256("magicdrop.factory.MagicDropCloneFactory")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 private constant MAGICDROP_FACTORY_STORAGE =
+        0xc982f4ee776d5a4a389c53cdba6d233ccdaf1824e9a3b0f1ea8fce06f767d800;
+
+    /*==============================================================
+    =                             EVENTS                           =
+    ==============================================================*/
+
+    event MagicDropFactoryInitialized();
+    event NewContractInitialized(
+        address contractAddress, address initialOwner, uint32 implId, TokenStandard standard, string name, string symbol
+    );
+
+    /*==============================================================
+    =                             ERRORS                           =
+    ==============================================================*/
+
+    error InitializationFailed();
+    error SaltAlreadyUsed();
+    error ContractAlreadyDeployed(address deployedAddress);
+    error RegistryAddressCannotBeZero();
+
+    /*==============================================================
+    =                          INITIALIZER                         =
+    ==============================================================*/
+
+    /// @notice Initializes the contract
+    /// @param initialOwner The address of the initial owner
+    /// @param registry The address of the registry contract
+    /// @dev This function can only be called once
+    function initialize(address initialOwner, address registry) public initializer {
+        if (registry == address(0)) {
+            revert RegistryAddressCannotBeZero();
+        }
+
+        _registry = MagicDropTokenImplRegistry(registry);
+        _initializeOwner(initialOwner);
+
+        emit MagicDropFactoryInitialized();
+    }
+
+    /*==============================================================
+    =                      PUBLIC WRITE METHODS                    =
+    ==============================================================*/
+
+    /// @notice Creates a new deterministic clone of a MagicDrop contract
+    /// @param name The name of the new contract
+    /// @param symbol The symbol of the new contract
+    /// @param standard The token standard of the new contract
+    /// @param initialOwner The initial owner of the new contract
+    /// @param implId The implementation ID
+    /// @param salt A unique salt for deterministic address generation
+    /// @return The address of the newly created contract
+    function createContractDeterministic(
```

Comment:
This might become obvious as I read more of the code but why do we need to know the contract's address?